framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml

```
@@ -0,0 +1,397 @@
+
+sWscAttrPrimaryDeviceType:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_PRIMARY_DEV_TYPE
+  data_length:
+    _type: uint16_t
+    _value: WSC_PRIMARY_DEV_TYPE_LENGTH
+  category_id:
+    _type: uint16_t 
+  oui:
+    _type: uint8_t
+    _length: [ WSC_PRIMARY_DEV_TYPE_OUI_LENGTH ]
+  sub_category_id:
+    _type: uint16_t 
+
+sWscAttrDeviceName:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_DEV_NAME
+  data_length:
+    _type: uint16_t
+    _value: WSC_MAX_NAME_LENGTH
+  data:
+    _type: char
+    _length: [ WSC_MAX_NAME_LENGTH ]
+
+sWscAttrRfBands:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_RF_BANDS
+  data_length:
+    _type: uint16_t
+    _value: 1
+  data:
+    _type: eWscRfBands
+
+sWscAttrAssociationState:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_ASSOC_STATE
+  data_length:
+    _type: uint16_t
+    _value: 1
+  data:
+    _type: eWscAssoc
+    _value: WSC_ASSOC_NOT_ASSOC
+
+sWscAttrDevicePasswordID:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_DEV_PASSWORD_ID
+  data_length:
+    _type: uint16_t
+    _value: 2
+  data:
+    _type: eWscValues16
+    _value: DEV_PW_PUSHBUTTON
+
+sWscAttrConfigurationError:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_CONFIG_ERROR
+  data_length:
+    _type: uint16_t
+    _value: 2
+  data:
+    _type: eWscValues16
+    _value: WSC_CFG_NO_ERROR
+
+sWscAttrOsVersion:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_OS_VERSION
+  data_length:
+    _type: uint16_t
+    _value: WSC_OS_VERSION_LENGTH
+  data:
+    _type: uint32_t
+    _value: 0x80000001
+
+sWscAttrVersionExtension:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_VENDOR_EXTENSION
+  data_length:
+    _type: uint16_t
+    _value: WSC_VENDOR_EXTENSIONS_LENGTH
+  data:
+    _type: uint8_t
+    _length: [ WSC_VENDOR_EXTENSIONS_LENGTH ]
+
+sWscAttrKeyWrapAuthenticator:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_KEY_WRAP_AUTH
+  data_length:
+    _type: uint16_t
+    _value: WSC_KEY_WRAP_AUTH_LENGTH
+  data:
+    _type: uint8_t
+    _length: [ WSC_KEY_WRAP_AUTH_LENGTH ]
+
+#encrypted attributes
+
+sWscAttrSsid:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_SSID
+  data_length:
+    _type: uint16_t
+    _value: WSC_MAX_NAME_LENGTH
+  data:
+    _type: char
+    _length: [ WSC_MAX_NAME_LENGTH ]
+
+sWscAttrAuthenticationType:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_AUTH_TYPE
+  data_length:
+    _type: uint16_t
+    _value: 2
+  data:
+    _type: uint16_t
+
+sWscAttrEncryptionType:
+  _type: struct
+  attribute_type:
+    _type: eWscAttributes
+    _value: ATTR_ENCR_TYPE
+  data_length:
+    _type: uint16_t
+    _value: 2
+  data:
+    _type: uint16_t
```

```suggestion
    _type: eWscEncr
```
Note that you can't do the same for sWscAttrEncryptionTypeFlags because that is a bitmask of eWscEncr values. The encryption type attribute, however, is only one.

---

fixed